### PR TITLE
Plagued lands

### PR DESCRIPTION
**Pull request by killerwife**
Quest 2118 fixup and some new core changes.

Required SQL for changes
DELETE FROM spell_script_target WHERE entry=9439;
INSERT INTO spell_script_target VALUES(9439, 1, 2164, 0);

UPDATE quest_template SET ReqSpellCast1=0 WHERE entry=2118;
UPDATE gameobject_template SET data2=10, data4=0 WHERE entry=109515;

DELETE FROM spell_dbc WHERE Id=9439; -- change to spell_template in open source for now
INSERT INTO spell_dbc(Id, SpellName, Effect1, EffectImplicitTargetA1, EffectBasePoints1, rangeIndex, SpellIconID, EquippedItemClass) VALUES
(9439, 'Bear Captured in Trap', 3, 38, 1, 12, 1, -1);

UPDATE creature_template SET FactionAlliance=35,FactionHorde=35 WHERE entry=11836;

EDIT: @cyberium @xfurry @cala

### CODE REVIEW

**Comment by xfurry** [comment]
Overall looks good from my side.
Also poke @cala as he wrote the script for this quest initially.

**Comment by killerwife** [comment]
Oh, so he is responsible for that hack :D

**Comment by cala** [comment]
It was written by someone else (the name should be in the commit), I just ported it.
It looks good to me too, but I must test it against Classic before because a few things changed between Classic and TBC (not only the spell removal IIRC). I should be able to answer by the end of the week or this WE if that sounds ok to you.

**Comment by killerwife** [comment]
@cala There are actually two removed spells, but quite frankly the second is redundant. (at least from my perspective) The grid search hack was just atrocious T_T (besides, this version works much more reliably) Tho let me know, should you find any bigger differencies on classic.

File: src/game/Spell.cpp [hunk 1]
```diff
@@ -4589,13 +4593,19 @@ SpellCastResult Spell::CheckCast(bool strict)
...
+                                            && pTarget->IsWithinDistInMap(worldObject, GetSpellMaxRange(srange)))
+                                            targetExplicit = (Creature*)pTarget;
+                                        else if (pTarget->IsWithinDistInMap(m_caster, GetSpellMaxRange(srange)))
                                             targetExplicit = (Creature*)pTarget;
                                     }
                                     else if (i_spellST->type == SPELL_TARGET_TYPE_CREATURE && pTarget->isAlive())
                                     {
                                         // always use spellMaxRange, in case GetLastRange returned different in a previous pass
-                                        if (pTarget->IsWithinDistInMap(m_caster, GetSpellMaxRange(srange)))
+                                        if (worldObject && (worldObject->GetTypeId() == TYPEID_GAMEOBJECT || worldObject->GetTypeId() == TYPEID_DYNAMICOBJECT)
```
Comment (kartoffelsup) [inline src/game/Spell.cpp]: Isn't this block and the block above exactly alike? Would it be possible to merge them into one?

Comment (killerwife) [inline src/game/Spell.cpp]: Now that you mention it, wasnt it the same before as well?

Comment (kartoffelsup) [inline src/game/Spell.cpp]: Looks like it :D

Comment (killerwife) [inline src/game/Spell.cpp]: After the guys look at it im gonna fix it. Im having a few days off :D

**Comment by cala** [comment]
Sorry, I'm busy, I should look at this against Classic pretty soon.

**Comment by killerwife** [comment]
Well dont worry, when ur done with it, I have a nice big fat gist of classic fixes waiting for u ^^

File: src/game/Spell.cpp (hunk 1 above)
Comment (Phatcat) [inline src/game/Spell.cpp]: You probably wanna look into this if you havn't yet.

File: src/scriptdev2/scripts/kalimdor/darkshore.cpp [hunk 2]
```diff
@@ -687,50 +688,35 @@ struct npc_rabid_bearAI : public ScriptedAI
         ScriptedAI::MoveInLineOfSight(pWho);
     }
 
-    void UpdateAI(const uint32 uiDiff) override
+    void SpellHit(Unit* pCaster, const SpellEntry* pSpell) override
```
Comment (Phatcat) [inline src/scriptdev2/scripts/kalimdor/darkshore.cpp]: no reason for that single trailing space I'm imagining :3

File: src/scriptdev2/scripts/kalimdor/darkshore.cpp [hunk 3]
```diff
@@ -687,50 +688,35 @@ struct npc_rabid_bearAI : public ScriptedAI
...
+                m_creature->GetMotionMaster()->MoveFollow(pCaster, PET_FOLLOW_DIST, PET_FOLLOW_ANGLE);
             }
-            else
-                m_uiCheckTimer -= uiDiff;
+            else                                // Something unexpected happened
+                m_creature->ForcedDespawn(1000);
+
+            // No need to check any more
+            m_uiCheckTimer = 0;
+            // Despawn after a while (delay guesswork)
```
Comment (Phatcat) [inline src/scriptdev2/scripts/kalimdor/darkshore.cpp]: what's the name of this quest, maybe I can help find out what the despawn time is.

Comment (killerwife) [inline src/scriptdev2/scripts/kalimdor/darkshore.cpp]: Its the name of the PR genius :D

**Comment by killerwife** [comment]
@Phatcat both ur comments done

**Comment by cala** [comment]
I should (finally) be able to test this against Classic within the week. Sorry for the long delay, life is full of surprises.

**Comment by cala** [comment]
Finally, all good for me. Successfully tested with Classic core, much smoother than the previous version. :+1:

**Comment by killerwife** [comment]
Awesome, well then, feel free to merge it then. We should finally have spell_dbc soon, so I will be able to insert the missing spell to TBC and WOTLK.

**Comment by xfurry** [comment]
Let's run a few tests on Wotlk with this first. Just to avoid any surprises :smiley:

**Comment by killerwife** [comment]
Asking for merge of this after DBC to SQL merge. Any objections?

**Comment by cyberium** [comment]
merged thanks

**Pull request closed without merging**